initfiles/sbin/hpcc_setenv.in

```diff
@@ -23,30 +23,46 @@
 ###<REPLACE>###
 
 function kill_process () {
-    SENTINEL=$1
-    PID=$2
-    TIMEOUT=$3
-    if [ -e $SENTINEL ]; then
-        rm -f $SENTINEL
+    local PID=$1
+    local PROCESS=$2
+    local TIMEOUT=$3
+    if [[ $# -eq 4 ]]; then
+      SENTINEL=$4
+      [[ -e $SENTINEL ]] && rm -f $SENTINEL
     fi
-    if [ -e $PID ]; then
-        pidwait_fn $PID $TIMEOUT
+    if [[ -e $PID ]]; then
+        pidwait_fn $PID $PROCESS $TIMEOUT
     fi
 }
 
 function pidwait_fn () {
-    PID=$1
-    WATCH_PID=`cat $PID`
-    TIMEOUT=$(($2*1000))
+    local PID=$1
+    local WATCH_PID=$( cat $PID )
+    local PROCESS=$2
+    local TIMEOUT=$(($3*1000))
+
+    if ps -p $WATCH_PID -o command= | grep -v "${PROCESS}" &>/dev/null ; then
+      exit 1
```

Comment:
hm, that means if WATCH_PID exists, but it's not ${PROCESS), then abort the script with an error?

If another process happened to have started to use PID, then that's okay.. it just means 'our' process has quit and OS has reused the PID # .. should just return success here I think.